src/main/java/at/grahsl/kafka/connect/mongodb/MongoDbSinkConnectorConfig.java

```diff
@@ -120,7 +123,52 @@ public MongoDbSinkConnectorConfig(Map<String, String> parsedConfig) {
     }
 
     public static ConfigDef conf() {
-        return new ConfigDef()
+        return new ConfigDef() {
+
+            <T> Validation<T> ensureValid(String name, Consumer<T> consumer) {
+                return new Validation<T>(name, consumer);
+            }
+
+            class Validation<T> {
+
+                private final String name;
+                private final Consumer<T> consumer;
+
+                Validation(String name, Consumer<T> consumer) {
+                    this.name = name;
+                    this.consumer = consumer;
+                }
+
+                Validation<T> unless(boolean condition) {
```

Comment:
codacy quality checks reports the following violation which is IMHO not a problem as such but maybe can be considered anyway. see https://app.codacy.com/app/hpgrahsl/kafka-connect-mongodb/pullRequest?prid=1396680

Reply:
https://github.com/941design/kafka-connect-mongodb/commit/0b94adcffd44dc43493bd7a154bd58fc3c7775e7